components/properties/src/script.rs

```diff
@@ -195,46 +206,22 @@ impl<'data> ScriptExtensions<'data> {
     /// Returns a [`UnicodeSet`] for the given [`Script`] which represents all
     /// code points for which `has_script` will return true.
     pub fn get_script_extensions_set(&self, script: Script) -> UnicodeSet {
+        let ranges_matching_script = self.trie.iter_ranges().filter(|cpm_range| {
```

Comment:
Suggestion: return `ranges_matching_script` directly, as an `impl Iterator<Item = CodePointMapRange>`, so that we can avoid creating the UnicodeSet if the user doesn't actually need the UnicodeSet.